LEAF_Request_Portal/View.php

```diff
@@ -21,88 +21,101 @@ public function __construct($db, $login)
         $form = new Form($this->db, $this->login);
-        if (!$form->hasReadAccess($recordID))
-        {
-            return 0;
-        }
-
-        $result = array();
-        require_once 'VAMC_Directory.php';
-        $dir = new VAMC_Directory;
 
-        $vars = array(':recordID' => $recordID);
-
-        $res = $this->db->prepared_query('SELECT * FROM action_history
-        									LEFT JOIN dependencies USING (dependencyID)
-                                            LEFT JOIN workflow_steps USING (stepID)
-        									LEFT JOIN actions USING (actionType)
-        									WHERE recordID=:recordID
-                                            ORDER BY time ASC', $vars);
-
-        foreach ($res as $tmp)
-        {
-            $packet = [];
-            $packet['time'] = $tmp['time'];
-            if($tmp['stepTitle'] != ''
-                && $tmp['dependencyID'] < 0) {
-                $packet['description'] = $tmp['stepTitle'] . ': ' . $tmp['actionText'];
-            }
-            else {
-                $packet['description'] = $tmp['description'] . ': ' . $tmp['actionText'];
-            }
-            if($tmp['description'] == ''
-                && $tmp['actionText'] == ''
-            ) {
-                $packet['description'] = 'Action';
-            }
-            $packet['comment'] = $tmp['comment'];
-            if ($tmp['userID'] != '')
+        if (!$form->hasReadAccess($recordID)) {
+            $return_value = array();
+        } else {
+            $result = array();
+            require_once 'VAMC_Directory.php';
+            $dir = new VAMC_Directory;
+
+            $vars = array(':recordID' => $recordID);
+            $sql1 = 'SELECT time, description, actionText, stepTitle,
+                        dependencyID, comment, userID
+                     FROM action_history
+                     LEFT JOIN dependencies USING (dependencyID)
+                     LEFT JOIN workflow_steps USING (stepID)
+                     LEFT JOIN actions USING (actionType)
+                     WHERE recordID=:recordID
+                     UNION
+                     SELECT timestamp, "Note Added", "N/A", "N/A",
+                        "N/A", note, userID
+                     FROM notes
+                     WHERE recordID = :recordID
+                     AND deleted IS NULL
+                     ORDER BY time ASC';
+
+            $res = $this->db->prepared_query($sql1, $vars);
+
+            foreach ($res as $tmp)
             {
-                $user = $dir->lookupLogin($tmp['userID']);
-                $name = isset($user[0]) ? "{$user[0]['Fname']} {$user[0]['Lname']}" : $tmp['userID'];
-                $packet['userName'] = $name;
+                $packet = [];
+                $packet['time'] = $tmp['time'];
+
+                if ($tmp['description'] == 'Note Added') {
+                    $packet['description'] = 'Note Added: ';
+                } else if ($tmp['description'] == '' && $tmp['actionText'] == '' ) {
```

Comment:
Instead of looking for empty strings, '',  you'll want to use the empty() function.  PHP8 is much more sensitive to the difference between '', 0, and null.